### PR TITLE
[MODORDERS-1120] - Support populating searchLocationId for correct tenant during order opening

### PR DESCRIPTION
**Pull request by imerabishvili**
## Purpose
https://folio-org.atlassian.net/browse/MODORDERS-1120

## Approach
Update `updateSearchLocations` logic to retrieve permanent locations by holdings from corresponding tenants

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/orders/utils/HelperUtils.java [hunk 1]
```diff
@@ -318,19 +317,30 @@ public static int getElectronicLocationsQuantity(List<Location> locations) {
       .sum();
   }
 
+  /*
+  Almost same as collectResultsOnSuccess with addition that each future itself returns a list and
```
Comment (azizbekxm) [inline src/main/java/org/folio/orders/utils/HelperUtils.java:321]: Let format this doc to be compatible with other in that file

File: src/main/java/org/folio/helper/PurchaseOrderLineHelper.java [hunk 2]
```diff
@@ -383,7 +380,7 @@ public String buildNewPoLineNumber(PoLine poLineFromStorage, String poNumber) {
    */
   public void updateEstimatedPrice(CompositePoLine compPoLine) {
     Cost cost = compPoLine.getCost();
-    cost.setPoLineEstimatedPrice(calculateEstimatedPrice(cost).getNumber().doubleValue());
+    cost.setPoLineEstimatedPrice(HelperUtils.calculateEstimatedPrice(cost).getNumber().doubleValue());
```
Comment (azizbekxm) [inline src/main/java/org/folio/helper/PurchaseOrderLineHelper.java:383]: This change is making code longer from my perspective. Previous version is more easier to read. Please consider in other places as well

File: src/main/java/org/folio/orders/utils/StreamUtils.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  public static <K, V> List<V> map(Collection<K> collection, Function<K, V> mapper) {
+    return collection.stream().map(mapper).toList();
+  }
+
+  public static <K, V> Set<V> mapToSet(Collection<K> collection, Function<K, V> mapper) {
+    return collection.stream().map(mapper).collect(Collectors.toSet());
+  }
+
+  public static <T, K> Map<K, T> list2map(Collection<T> collection, Function<T, K> toKey) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/orders/utils/StreamUtils.java:42]: if we have already in contract of this class mapToSet, so better to name this method as listToMap

File: src/main/java/org/folio/helper/PurchaseOrderLineHelper.java (hunk 2 above)
Comment (imerabishvili) [reply src/main/java/org/folio/helper/PurchaseOrderLineHelper.java:384]: done

**Review by SerhiiNosko — APPROVED** [review]

**Review by azizbekxm — APPROVED** [review]

**Pull request merged**